src/include/configuration/configuration_manager.h

```diff
@@ -0,0 +1,137 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// configuration_manager.h
+//
+// Identification: src/include/configuration/configuration_manager.h
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include <gflags/gflags.h>
+#include "type/types.h"
+#include "type/ephemeral_pool.h"
+#include "common/exception.h"
+#include "catalog/config_catalog.h"
+#include "concurrency/transaction_manager_factory.h"
```

Comment:
It seems that you don't need to include config_catalog and transaction_manager_factory here. You can include them in the .cpp file to reduce coupling of the code base. You should also forward declare ephemeral_pool since it's only used with unique_ptr.